analyzers/src/SonarAnalyzer.CSharp/Rules/InfiniteRecursion.cs

```diff
@@ -65,6 +65,14 @@ protected override void Initialize(SonarAnalysisContext context)
                     }
                 },
                 SyntaxKind.PropertyDeclaration);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var @operator = (OperatorDeclarationSyntax)c.Node;
+                    CheckForNoExitMethod(c, (CSharpSyntaxNode)@operator.Body ?? @operator.ExpressionBody, @operator.OperatorToken);
```

Comment:
here you check if the operator has a body - from the tests I only saw arrow methods - am I wrong?

to be explicit: please have block methods as well in the tests, to cover this logic